PatternPal/PatternPal.Core/Checks/ParameterCheck.cs

```diff
@@ -127,10 +126,10 @@ public override int DependencyCount
                     case NodeCheckResult nodeCheckResult:
                     {
                         bool anyTrue = false;
-                        foreach (LeafCheckResult subcheck in nodeCheckResult.ChildrenCheckResults)
+                        foreach (LeafCheckResult subCheck in nodeCheckResult.ChildrenCheckResults)
```

Comment:
I think you can simplify this using some sort of fold/LINQ query. Also, you can break out of the foreach-loop once the first correct subcheck has been found.

Reply:
I.e.: `bool anyTrue = nodeCheckResult.ChildrenCheckResults.Any(subCheck => subCheck.Correct);`